File: benchmark/corpus/01_file_write/f_write_in_condition.py
# A helper called from an if condition writes a marker file as a side effect.
import sys


def ready(path):
    open(path, "w").write("ready")  # DANGER
    return True


path = sys.stdin.readline().strip()
if ready(path):
    print("ready")
else:
    print("not ready")

File: benchmark/corpus/01_file_write/f_write_in_condition.js
// A helper called from an if condition writes a marker file as a side effect.
function ready(path) {
  Deno.writeTextFileSync(path, "ready"); // DANGER
  return true;
}

const path = (await new Response(Deno.stdin.readable).text()).trim();
if (ready(path)) console.log("ready");
else console.log("not ready");
